sinks/channel/channel_sink.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package channel
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/hashicorp/eventlogger"
+)
+
+// ChannelSink is a sink node which sends
+// the event to a channel
+type ChannelSink struct {
+	mu sync.Mutex
+
+	eventChan chan *eventlogger.Event
+}
+
+var _ eventlogger.Node = &ChannelSink{}
+
+// newChannelSink creates a ChannelSink
+func NewChannelSink(c chan *eventlogger.Event) (*ChannelSink, error) {
+	if c == nil {
+		return nil, errors.New("missing event channel")
+	}
+
+	return &ChannelSink{
+		eventChan: c,
+	}, nil
+}
+
+// Process sends the event on a channel
+// Returns a nil event as this is a leaf node
+func (c *ChannelSink) Process(ctx context.Context, e *eventlogger.Event) (*eventlogger.Event, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.eventChan <- e
```

Comment:
This can block indefinitely, what do you think about making this able to use the context to cancel this operation, like so:
```suggestion
	select {
	case c.eventChan <- e:
	case <-ctx.Done():
		return nil, ctx.Err()
	}
```

Reply:
Maybe something like:

```
select {
    case c.eventChan <- e:
    case <-ctx.Done():
        return nil, ctx.Err()
    case <-time.After(1 * c.errOnTimeoutDuration):
        return nil, fmt.Errorf("%s: chan write timeout", op)
}
```